### DISCUSSION TITLE
Layout files

### DISCUSSION THREAD

**Discussion by Dekker500** [hbstack/.github#11] [Q&A] [answered]
OK, I am struggling here...

Where can we customize the layout for individual elements, such as blog posts, or even the main/front page?

This is so different from 'core' Hugo that I am having a hard time... Love the pluggable aspect, but each module needs the ability to customize the output.

Is there documentation on how to **extend** or **style** existing modules? The modules provide the core functionality, but the layout should be completely style-able otherwise it is very limiting...

**Comment by razonyang** [comment, marked as answer]
It's no different with Hugo if you follow the [look up order](https://gohugo.io/templates/lookup-order/).

But those layouts aren't documented, you can find it within the source code.

> Please note that there is no guarantee that those layouts are stable, they will be changed in feature release (BREAKING CHANGES).

EDIT, this may help https://hb.hugomods.com/en/docs/advanced/

**Reply by Dekker500** [reply]
Marked as answer, but still quite difficult to determine where to place things.

e.g. one would think that adding text to "content/index.md" would result in text on the front/main page... but it does not. Investigation continues. Just the simple task of determining what the partial is called, and where it should be located, makes customizing the look-and-feel a developer chore.

Will see what I can come up with.

**Reply by razonyang** [reply]
Agree with you, it's hard to find which layout/template is being using.

Btw, the docs are almost done, you may be interested in this https://hb.hugomods.com/en/docs/develop/module/.

**Comment by razonyang** [comment]
Closing per above comments
